module/zfs/sa.c

```diff
@@ -1017,8 +1020,12 @@ sa_setup(objset_t *os, uint64_t sa_obj, sa_attr_reg_t *reg_attrs, int count,
 	mutex_init(&sa->sa_lock, NULL, MUTEX_DEFAULT, NULL);
```

Comment:
Instead of creating a subclass per lock why don't we go ahead and disable lockdep checking for the `sa_lock`.  The following change should be sufficient.

```diff
index 4999fef34..f718e7662 100644
--- a/module/zfs/sa.c
+++ b/module/zfs/sa.c
@@ -1014,7 +1014,7 @@ sa_setup(objset_t *os, uint64_t sa_obj, sa_attr_reg_t *reg_attrs, int count,
        }
 
        sa = kmem_zalloc(sizeof (sa_os_t), KM_SLEEP);
-       mutex_init(&sa->sa_lock, NULL, MUTEX_DEFAULT, NULL);
+       mutex_init(&sa->sa_lock, NULL, MUTEX_NOLOCKDEP, NULL);
        sa->sa_master_obj = sa_obj;
 
        os->os_sa = sa;
```